trin-core/src/portalnet/overlay.rs

```diff
@@ -187,6 +188,44 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
             .collect()
     }
 
+    /// Returns an iterator over all the entries in the routing table.
+    pub fn table_entries(&self) -> Vec<(NodeId, Enr, NodeStatus)> {
+        self.kbuckets
+            .write()
+            .iter()
+            .map(|entry| {
+                (
+                    *entry.node.key.preimage(),
+                    entry.node.value.enr().clone(),
+                    entry.status,
+                )
+            })
+            .collect()
+    }
+
+    /// Returns vector of all ENR node IDs of nodes currently contained in the routing table mapped to JSON Value.
+    pub fn routing_table_info(&self) -> Value {
+        let buckets: Vec<(String, String, String)> = self
+            .table_entries()
+            .iter()
+            .map(|(node_id, enr, node_status)| {
+                (
+                    node_id.to_string(),
```

Comment:
I've been working on a branch doing some jsonrpc cleanup. `node_id.to_string()` returns something like `0xabcd..1234`. No doubt, this is nice and readable, but I've been updating it to `format!("0x{}", hex::encode(self.discv5.local_enr().node_id().raw()))` b/c I feel like we should be returning the actual node id rather than the abbrev.